src/pveclib/vec_int512_ppc.h

```diff
@@ -0,0 +1,2167 @@
+ *
+ *
+ * \todo General discussion of static and dynamic libraries should be
+ * moved (eventually) to pveclibmaindox.h. While specifics of
+ * multiple quadword precision integer multiply should remain here.
+ *
+ * Until recently (as of v1.0.3) PVECLIB operations where
+ * <B>static inline</B> only. This was reasonable as most operations
+ * where small (a one to a few vector instructions).
+ * This offered the compiler opportunity for;
+ * - Better register allocation.
+ * - Identifying common subexpressions and factoring them across operations instances.
+ * - Better instruction scheduling across operations.
+ *
+ * Even then a few operations
+ * (quadword multiply, BCD multiply, BCD <-> binary conversions,
+ * and some POWER8/7 implementations of POWER9 instructions)
+ * where getting uncomfortably large (10s of instructions).
+ * But it was the multiple quadword precision operations that
+ * forced the issue as they can run to 100s and sometimes 1000s of
+ * instructions.
+ *
+ * \subsection i512_libary_issues_0_0_0 Building Multi-platform Libraries
+ *
+ * Building libraries of compiled binaries is not that difficult.
+ * The challenge is effectively supporting multiple processor
+ * (POWER7/8/9) platforms as many PVECLIB operations have different
+ * implementations for each platform.
+ * This is especially evident on the multiply integer word, doubleword,
+ * and quadword operations (see; vec_muludq(), vec_mulhuq(), vec_mulluq(),
+ * vec_vmuleud(), vec_vmuloud(), vec_msumudm(), vec_muleuw(),
+ * vec_mulouw()).
+ *
+ * This is dictated by both changes in the PowerISA and in
+ * the micro-architecture as it evolved across processor generations.
+ * So an implementation to run on a POWER7 is necessarily restricted to
+ * the instructions of PowerISA 2.06.
+ * But if we are running on a POWER9, leveraging new instructions from
+ * PowerISA 3.0 can yield better performance than the POWER7
+ * compatible implementation. When we are dealing with larger operations
+ * (10s and 100s of instructions) the compiler can schedule instruction
+ * sequences based on the platform (-mtune=) for better performance.
+ *
+ * So we need to deliver multiple implementations for some operations
+ * and we need to provide mechanisms to select a specific platform
+ * implementation statically at compile/build or dynamically at
+ * runtime. First we need to compile multiple version of these
+ * operations, as unique functions, each with a different effective
+ * target platform (-mcpu=).
+ *
+ * Obviously creating multiple source files implementing the same large
+ * operation, each supporting a specific target platform,
+ * is a possibility.
+ * However this could cause maintenance problems where changes to a
+ * operation must be coordinated across multiple source files.
+ * This is also inconsistent with the current PVECLIB coding style
+ * where all where an operations complete implementation, including
```

Comment:
nit: `operation's`.

Reply:
Ok, I think I can simplify the word salad to:

 * However this could cause maintenance problems where changes to a
 * operation must be coordinated across multiple source files.
 * This is also inconsistent with the current PVECLIB coding style
 * where a file contains an operations complete implementation,
 * including documentation and platform implementation variants.

Reply:
No negative responses, so resolved